docs/docsite/rst/dev_guide/migrating_roles.rst

```diff
@@ -0,0 +1,345 @@
+      │   └── .../
+      ├── README.md
+      ├── roles/
+      │   ├── role1/
+      │   ├── role2/
+      │   └── .../
+      ├── playbooks/
+      │   ├── files/
+      │   ├── vars/
+      │   ├── templates/
+      │   └── tasks/
+      └── tests/
+
+A collection can contain one or more roles in the :file:`roles/` directory and these are almost identical to standalone roles with the following exceptions:
+
++----------------------------------------+------------------------------------------------------------------------------+
+| standalone role                        | collection role                                                              |
++========================================+==============================================================================+
+| :file:`library/` with custom modules   | - Move to :file:`plugins/modules/`                                           |
+|                                        | - Use :abbr:`FQCN (Fully Qualified Collection Name)` external to collection  |
++----------------------------------------+------------------------------------------------------------------------------+
+|``ansible.module_utils`` in Python      + Use :abbr:`FQCN (Fully Qualified Collection Name)` for Python imports        |
++----------------------------------------+------------------------------------------------------------------------------+
+| Plugins in :file:`PLUGINTYPE_plugins/` + - Move to :file:`plugins/PLUGINTYPE/`  (such as :file:`plugins/filter/`)     |
+|                                        | - Use :abbr:`FQCN (Fully Qualified Collection Name)` external to collection  |
++----------------------------------------+------------------------------------------------------------------------------+
+| :file:`module_utils/`                  + - Move to :file:`plugins/module_utils/`                                      |
+|                                        | - Use :abbr:`FQCN (Fully Qualified Collection Name)` external to collection  |
++----------------------------------------+------------------------------------------------------------------------------+
+
+.. note::
+
+	 In standalone roles, some of the plugin directories referenced their plugin types in the plural sense; this is not the case in collections.
+
+Migrating a role to a collection
+================================
+
+To migrate from a standalone role to a collection role:
+
+
+1. Create a collection. You need a `Galaxy namespace <https://galaxy.ansible.com/docs/contributing/namespaces.html>`_ to import this collection to Galaxy.
+
+.. code-block:: bash
+
+  $ ansible-galaxy collection init mynamespace.mycollection
+
+This creates the collection directory structure.
+
+2. Copy the standalone role directory into the :file:`roles/` subdirectory of the collection.
+
+.. code-block:: bash
+
+  $ mkdir mynamespace/mycollection/roles/my_role/
+  $ cp -r /path/to/standalone/role/mynamespace/my_role/\* mynamespace/mycollection/roles/my_role/
```

Comment:
note that this might require a rename of role since naming rules in collection differ from 'non collection roles'